support-frontend/assets/helpers/user/user.ts

```diff
@@ -143,43 +137,45 @@ const init = (
 	}
 
 	if (windowHasUser) {
-		dispatch(setId(window.guardian.user.id));
-		dispatch(setEmail(window.guardian.user.email));
-		dispatch(setDisplayName(window.guardian.user.displayName));
-		dispatch(setFirstName(window.guardian.user.firstName));
-		dispatch(setLastName(window.guardian.user.lastName));
-		dispatch(
-			setFullName(
-				`${window.guardian.user.firstName} ${window.guardian.user.lastName}`,
-			),
-		);
+		const { id, email, displayName, firstName, lastName, address4 } =
+			windowHasUser;
+
+		id && dispatch(setId(id));
+		email && dispatch(setEmail(email));
+		displayName && dispatch(setDisplayName(displayName));
+		firstName && dispatch(setFirstName(firstName));
+		lastName && dispatch(setLastName(lastName));
+		dispatch(setFullName(`${firstName} ${lastName}`));
+
 		// default value from Identity Billing Address, or Fastly GEO-IP
-		dispatch(
-			setStateField(
-				window.guardian.user.address4 || window.guardian.geoip.stateCode,
-			),
-		);
+		if (address4) {
+			dispatch(setStateField(address4));
+		} else {
+			window.guardian.geoip?.stateCode &&
+				dispatch(setStateField(window.guardian.geoip.stateCode));
+		}
```

Comment:
now if the stateCode and address4 are missing it won't dispatch anything, before it would have dispatched undefined or null - is this change important?

Reply:
I don't believe so because in our user reducer [here](https://github.com/guardian/support-frontend/blob/c1584b68f25f1a9c87eb463fbb940f88df146b06/support-frontend/assets/helpers/user/userReducer.ts#L47) if the type doesn't match one of the predefined strings then we'd return the default state